app/graphql/mutations/destroy/user.rb

```diff
@@ -0,0 +1,16 @@
+module Mutations
+  module Destroy
+    class User < ::Mutations::Destroy::Base
+      argument :id, !types.Int, 'ID of the user getting updated or deleted'
+
+      description 'Update a member status'
+      type ::Types::UserType
+
+      def call(_, args, ctx)
+        user = ::User.find args[:id]
+        destroy_generic user, args, ctx if user.participations.empty?
+        raise Pundit::NotAuthorizedError, "Can't remove someone that have participated!"
```

Comment:
We should use the Pundit policy for users to manage these errors. `create_generic` and `update_generic` have examples of how to access the Pundit policy from GraphQL.